models/gans/wgan.py

```diff
@@ -1,12 +1,181 @@
 # TODO: Implement the wgan architecture (issue #6)
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torch.autograd import Variable
+from torch import autograd
+import time as t
+import matplotlib.pyplot as plt
+plt.switch_backend('agg')
+import os
+from itertools import chain
+from torchvision import utils as utils
+from utils.logging import logging
 
-class WGAN:
-    pass
+SAVE_PER_TIMES = 100
 
 
-class Generator:
-    pass
+class WGAN(object):
+    def __init__(self, args):
+        print("WGAN_GradientPenalty init model.")
+        self.G = Generator(args.channels)
+        self.D = Discriminator(args.channels)
+        self.C = args.channels
 
+        # Check if cuda is available
+        self.check_cuda(args.cuda)
 
-class Discriminator:
-    pass
\ No newline at end of file
+        # WGAN values from paper
+        self.learning_rate = 1e-4
+        self.b1 = 0.5
+        self.b2 = 0.999
+        self.batch_size = 64
+
+        # WGAN_gradient penalty uses ADAM
```

Comment:
Should this be in the GANTrainer instead?

Reply:
I thing that if the trainer classes are initialized w/ DataLoaders, the batch_size would have to be set before initializing those classes.

Reply:
I left a bunch of code that could be transferred into the trainer